drivers/regulator/regulator_npm6001.c

```diff
@@ -0,0 +1,680 @@
+		break;
+	case NPM6001_SOURCE_LDO1:
+		stop_reg = NPM6001_TASKS_STOP_LDO1;
+		break;
+	default:
+		return 0;
+	}
+
+	return i2c_reg_write_byte_dt(&pconfig->i2c, stop_reg, 1U);
+}
+
+static int regulator_npm6001_get_error_flags(const struct device *dev,
+					     regulator_error_flags_t *flags)
+{
+	const struct regulator_npm6001_config *config = dev->config;
+	const struct regulator_npm6001_pconfig *pconfig = config->p->config;
+	uint8_t oc_reg, val;
+	int ret;
+
+	*flags = 0U;
+
+	/* read thermal warning */
+	ret = i2c_reg_read_byte_dt(&pconfig->i2c, NPM6001_EVENTS_THWARN, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (val != 0U) {
+		/* clear thermal warning */
+		ret = i2c_reg_write_byte_dt(&pconfig->i2c,
+					    NPM6001_EVENTS_THWARN, 0U);
+		if (ret < 0) {
+			return ret;
+		}
+
+		*flags |= REGULATOR_ERROR_OVER_TEMP;
+	}
+
+	/* read overcurrent event */
+	switch (config->source) {
+	case NPM6001_SOURCE_BUCK0:
+		oc_reg = NPM6001_EVENTS_BUCK0OC;
+		break;
+	case NPM6001_SOURCE_BUCK1:
+		oc_reg = NPM6001_EVENTS_BUCK1OC;
+		break;
+	case NPM6001_SOURCE_BUCK2:
+		oc_reg = NPM6001_EVENTS_BUCK2OC;
+		break;
+	case NPM6001_SOURCE_BUCK3:
+		oc_reg = NPM6001_EVENTS_BUCK3OC;
+		break;
+	default:
+		return 0;
+	}
+
+	ret = i2c_reg_read_byte_dt(&pconfig->i2c, oc_reg, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (val != 0U) {
+		/* clear overcurrent event */
+		ret = i2c_reg_write_byte_dt(&pconfig->i2c, oc_reg, 0U);
+		if (ret < 0) {
+			return ret;
+		}
+
+		*flags |= REGULATOR_ERROR_OVER_CURRENT;
+	}
+
+	return 0;
+}
+
+static int regulator_npm6001_init(const struct device *dev)
+{
+	const struct regulator_npm6001_config *config = dev->config;
+
+	regulator_common_data_init(dev);
+
+	if (!device_is_ready(config->p)) {
+		return -ENODEV;
+	}
+
+	return regulator_common_init_enable(dev);
+}
+
+static int regulator_npm6001_common_init(const struct device *dev)
+{
+	const struct regulator_npm6001_pconfig *config = dev->config;
+	int ret;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		return -ENODEV;
+	}
+
+	/* always select BUCK3 DAC (does not increase power consumption) */
+	ret = i2c_reg_write_byte_dt(&config->i2c, NPM6001_BUCK3SELDAC, 1U);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* configure pad properties */
+	ret = i2c_reg_write_byte_dt(&config->i2c, NPM6001_BUCKMODEPADCONF,
+				    config->buck_pad_val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&config->i2c, NPM6001_PADDRIVESTRENGTH,
+				    config->pad_val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static const struct regulator_driver_api api = {
+	.enable = regulator_npm6001_enable,
+	.disable = regulator_npm6001_disable,
+	.count_voltages = regulator_npm6001_count_voltages,
+	.list_voltage = regulator_npm6001_list_voltage,
+	.set_voltage = regulator_npm6001_set_voltage,
+	.get_voltage = regulator_npm6001_get_voltage,
+	.set_mode = regulator_npm6001_set_mode,
+	.get_error_flags = regulator_npm6001_get_error_flags,
```

Comment:
Is there a good reason to prefix them with regulator_ and not just npm6001_?

Reply:
just a convention used in many other drivers (`class_name_func`), it leads to long names, so I'm happy to change it in a future PR.

Reply:
IMO it could be just `reg_npm6001_`, others like internal `regulator_npm6001_buck0123_get_voltage()` just `npm6001_buck0123_get_voltage()`, it would better fits with your style in the file.